### ISSUE TITLE
mAP is much lower in  YOLOv3 INT8 model

### ISSUE THREAD

**Issue by SAIL-LAB155** [jkjung-avt/tensorrt_demos#381]
**Hi, this is the eval result of YOLOv3 TRT model with FP16**
![Screenshot from 2021-03-24 10-30-44](https://user-images.githubusercontent.com/56857281/112245640-3fe1ba00-8c8c-11eb-9303-3a07d287286b.png)
**This is  the eval result of YOLOv3 TRT model with INT8**
![Screenshot from 2021-03-24 10-26-44](https://user-images.githubusercontent.com/56857281/112245711-5e47b580-8c8c-11eb-8056-8fbb2c436f71.png)
**And my dataset is about 4k images, and I use  the val dataset for calibration which is about 400 images.
Do you know why mAP drops sharply? Thx.**

**Comment by jkjung-avt** [comment]
No, I don't have any idea about what you've described.  Maybe you could look at a few example images (from your val set) to see why the INT8 engine performs poorly.

**Comment by Sean-hku** [comment]
Do u mean I should use part of val set rather than whole val set?

**Comment by jkjung-avt** [comment]
No, I suggest running "trt_yolo.py" and visualize the resulting detections (bounding boxes).  See whether the bad mAP is due to missing detections or false positives or else, and try to figure out the next debugging step from there.

**Comment by Sean-hku** [comment]
Thank you for your quick reply. I will try it.

**Comment by SAIL-LAB155** [comment]
I find there are lots of missing detections in int8 models. What can I do to improve the mAP?
Thx

**Comment by jkjung-avt** [comment]
Honestly, this would be difficult to debug.  You could try:

1. set a lower "conf_th" (confidence threshold) to see if you get better detection results,

    https://github.com/jkjung-avt/tensorrt_demos/blob/d6ae0d7cc7fc356054c896eb5f9b0873d5d5ab6a/trt_yolo.py#L101

2. check if NVIDIA's new [Polygraphy](https://github.com/NVIDIA/TensorRT/tree/master/tools/Polygraphy) tool could be used to debug the TensorRT INT8 model.  You could refer to https://github.com/jkjung-avt/tensorrt_demos/issues/315#issuecomment-805419258 for more information.

**Comment by jkjung-avt** [comment]
I would also suggest you to try use as many images as possible (e.g. use your whole training set and test set) for INT8 calibration.  If that helps to improve mAP of the INT8 engine, then you know the problem lies in calibration dataset.
